cli/add.go

```diff
@@ -134,35 +137,97 @@ func logEnvs() error {
 	return nil
 }
 
+// readMax reads maximum the given amount of bytes
+func readMax(reader io.Reader, maxSize uint) (string, error) {
+	p := make([]byte, maxSize)
+	if _, err := reader.Read(p); err != nil && err != io.EOF {
+		return "", err
+	}
+	var r []byte
+	for _, b := range p {
+		if b != 0 {
+			r = append(r, b)
+		}
+	}
+	return string(r), nil
+}
+
+var errTimeout = errors.New("timeout")
+
+// readMaxWithTimeout reads maximum a given amount of time using readMax
+func readMaxWithTimeout(reader io.Reader, maxSize uint, timeout time.Duration) (string, error) {
+	valueChan := make(chan string)
+	errChan := make(chan error)
+
+	go func() {
+		v, err := readMax(reader, maxSize)
+		if err != nil {
+			errChan <- err
+			return
+		}
+		valueChan <- v
+	}()
+
+	var value string
+	var err error
+	select {
+	case value = <-valueChan:
+	case err = <-errChan:
+	case <-time.After(timeout):
+		err = errTimeout
+	}
+	return value, err
+}
+
 func add(c *cli.Context) error {
-	log.Debugln("[ENVMAN] - Work path:", envman.CurrentEnvStoreFilePath)
+	log.Debugln("[ENVMAN] Work path:", envman.CurrentEnvStoreFilePath)
 
 	key := c.String(KeyKey)
 	expand := !c.Bool(NoExpandKey)
 	replace := !c.Bool(AppendKey)
 	skipIfEmpty := c.Bool(SkipIfEmptyKey)
 
 	var value string
-	if stdinValue != "" {
-		value = stdinValue
-	} else if c.IsSet(ValueKey) {
+
+	// read piped stdin value
+	info, err := os.Stdin.Stat()
+	if err != nil {
+		log.Fatalf("[ENVMAN] Failed to get standard input FileInfo: %s", err)
+	}
+	if info.Mode()&os.ModeNamedPipe == os.ModeNamedPipe {
+		// max BASH env size: 20kB
+		var err error
+		value, err = readMaxWithTimeout(os.Stdin, 20*1024, 1*time.Second)
```

Comment:
instead of this function, please set the `os.Stdin.SetReadDeadline` so it will be handled internally